dashboard/api_edx_cache.py

```diff
@@ -16,6 +16,45 @@
 
 
 class CachedEdxUserData:
+    """Represents all edx data related to a User"""
+    # pylint: disable=too-many-instance-attributes
+
+    enrollments = None
+    certificates = None
+    current_grades = None
+    raw_enrollments = None
+    raw_certificates = None
+    raw_current_grades = None
+
+    def __init__(self, user, program=None, include_raw_data=False):
+        """
+        Fetches the given User's edx data and sets object properties
+
+        Args:
+            user (User): a User object
+            program (Program): an optional Program to filter on
+            include_raw_data (bool): Set to True if raw edx data is needed (in addition to actual edX objects)
+        """
+        self.user = user
+        self.program = program
+        self.fetch_and_set_edx_data()
+        if include_raw_data:
+            self.fetch_and_set_raw_data()
+
+    def fetch_and_set_edx_data(self):
+        """Fetches edx data and sets object properties"""
+        self.enrollments = models.CachedEnrollment.get_edx_data(self.user, program=self.program)
+        self.certificates = models.CachedCertificate.get_edx_data(self.user, program=self.program)
+        self.current_grades = models.CachedCurrentGrade.get_edx_data(self.user, program=self.program)
+
+    def fetch_and_set_raw_data(self):
+        """Fetches raw cached data and sets object properties"""
+        self.raw_enrollments = models.CachedEnrollment.get_raw_data(self.user, program=self.program)
+        self.raw_certificates = models.CachedCertificate.get_raw_data(self.user, program=self.program)
+        self.raw_current_grades = models.CachedCurrentGrade.get_raw_data(self.user, program=self.program)
```

Comment:
each `edx_api_client` object has all the JSON in a dictionary where the keys are the `course_key`.
for each object there is a method or property to access the values of the dictionary (like `Enrollments.enrolled_courses`.
I think that then it might get a bit too complicated. It is up to you if it is worth it.